src/java_bytecode/java_object_factory.cpp

```diff
@@ -386,6 +386,7 @@ void java_object_factoryt::gen_nondet_array_init(
 
   side_effect_exprt java_new_array(ID_java_new_array, expr.type());
   java_new_array.copy_to_operands(length_sym_expr);
+  java_new_array.set(ID_length_upper_bound, max_length_expr);
```

Comment:
`length_sym_expr`? No, the idea is that `length_sym_expr` is the actual size of the array and `max_length_expr` is the upper bound on that length. The actual size is used to initialise the array with NONDET members, and the bound is used to allocate the storage and provide a bound on the initialisation loop.